apis/config/v1alpha1/projectconfig_types.go

```diff
@@ -7,6 +7,59 @@ import (
 )
 
+// BuiltInCertManagement is the configuration for the built-in facility to generate and rotate
+// TLS client and serving certificates for all Tempo services and internal clients except
+// for the tempo-gateway.
+type BuiltInCertManagement struct {
+	// CACertValidity defines the total duration of the CA certificate validity.
+	CACertValidity metav1.Duration `json:"caValidity,omitempty"`
+	// CACertRefresh defines the duration of the CA certificate validity until a rotation
+	// should happen. It can be set up to 80% of CA certificate validity or equal to the
+	// CA certificate validity. Latter should be used only for rotating only when expired.
+	CACertRefresh metav1.Duration `json:"caRefresh,omitempty"`
+	// CertValidity defines the total duration of the validity for all Tempo certificates.
+	CertValidity metav1.Duration `json:"certValidity,omitempty"`
+	// CertRefresh defines the duration of the certificate validity until a rotation
+	// should happen. It can be set up to 80% of certificate validity or equal to the
+	// certificate validity. Latter should be used only for rotating only when expired.
+	// The refresh is applied to all Tempo certificates at once.
+	CertRefresh metav1.Duration `json:"certRefresh,omitempty"`
+	// Enabled defines to flag to enable/disable built-in certificate management feature gate.
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+// FeatureGates is the supported set of all operator feature gates.
+type FeatureGates struct {
+	// BuiltInCertManagement enables the built-in facility for generating and rotating
+	// TLS client and serving certificates for all Tempo component services and internal clients except
+	// for the tempo-gateway, In detail all internal Tempo HTTP and GRPC communication is lifted
+	// to require mTLS. For the tempo-gateway you need to provide a secret with or use the `ServingCertsS
+	// ervice`
+	// on OpenShift:
+	// - `tls.crt`: The TLS server side certificate.
+	// - `tls.key`: The TLS key for server-side encryption.
+	// In addition each service requires a configmap named as the MicroService CR with the
+	// suffix `-ca-bundle`, e.g. `tempo-dev-ca-bundle` and the following data:
+	// - `service-ca.crt`: The CA signing the service certificate in `tls.crt`.
+	BuiltInCertManagement BuiltInCertManagement `json:"builtInCertManagement,omitempty"`
+	// HTTPEncryption enables TLS encryption for all HTTP Microservices services.
+	// Each HTTP service requires a secret named as the service with the following data:
+	// - `tls.crt`: The TLS server side certificate.
+	// - `tls.key`: The TLS key for server-side encryption.
+	// In addition each service requires a configmap named as the Microservices CR with the
+	// suffix `-ca-bundle`, e.g. `tempo-dev-ca-bundle` and the following data:
+	// - `service-ca.crt`: The CA signing the service certificate in `tls.crt`.
+	HTTPEncryption bool `json:"httpEncryption,omitempty"`
```

Comment:
I think the comment should make it clear which services are protected and which not (public vs internal)